src/asset.ts

```diff
@@ -4,17 +4,72 @@ import { HttpMethod } from './types';
 import { buildData } from './util';
 
 export default class Asset extends AinftBase {
+  async getNftContractBySymbol(appId: string, symbol: string) {
+    const timestamp = Date.now();
+    const query = { appId, symbol: encodeURIComponent(symbol) };
+    const trailingUrl = 'nft';
+    const data = buildData(
+      HttpMethod.GET,
+      `/asset/${trailingUrl}`,
+      timestamp,
+      query
+    );
+    const signature = this.signData(data);
+    return axios
+      .get(`${this.baseUrl}/${trailingUrl}`, {
+        params: query,
+        headers: {
+          'X-AINFT-Date': timestamp,
+          Authorization: `AINFT ${signature}`,
+        },
+      })
+      .then((res) => res.data.data)
+      .catch((e) => {
+        throw e.response.data;
+      });
+  }
+
+  async getNft(
+    appId: string,
+    chainId: string,
+    contractAddress: string,
+    tokenId: string,
+  ) {
+    const timestamp = Date.now();
+    const query = { appId, contractAddress, tokenId };
+    const trailingUrl = `nft/${chainId}`;
+    const data = buildData(
+      HttpMethod.GET,
+      `/asset/${trailingUrl}`,
+      timestamp,
+      query
+    );
+    const signature = this.signData(data);
+    return axios
+      .get(`${this.baseUrl}/${trailingUrl}`, {
+        params: query,
+        headers: {
+          'X-AINFT-Date': timestamp,
+          Authorization: `AINFT ${signature}`,
+        },
+      })
+      .then((res) => res.data.data)
+      .catch((e) => {
+        throw e.response.data;
+      });
+  }
+
   async getUserNftList(
     appId: string,
     chainId: string,
     ethAddress: string,
-    contractAddress?: string
+    contractAddress: string,
```

Comment:
It would be good to discuss whether to receive the contract address as an option.

Reply:
🙆‍♀️ discussed on the other pr & updated to keep this optional